core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -321,17 +326,25 @@ protected boolean shouldUpdateStreams(long currentBlockId) {
 
   /**
    * Closes or cancels {@link #mCurrentCacheStream}.
-   *
-   * @throws IOException if the close or cancel fails
    */
-  private void closeOrCancelCacheStream() throws IOException {
+  private void closeOrCancelCacheStream() {
     if (mCurrentCacheStream == null) {
       return;
     }
-    if (mCurrentCacheStream.remaining() == 0) {
-      mCurrentCacheStream.close();
-    } else {
-      mCurrentCacheStream.cancel();
+    try {
+      if (mCurrentCacheStream.remaining() == 0) {
+        mCurrentCacheStream.close();
+      } else {
+        mCurrentCacheStream.cancel();
+      }
+    } catch (IOException e) {
+      if (e.getCause() instanceof BlockDoesNotExistException) {
+        LOG.info("Block {} does not exist when being cancelled.", getCurrentBlockId());
+      } else if (e.getCause() instanceof InvalidWorkerStateException) {
+        LOG.info("Block {} has invalid worker state when being cancelled.", getCurrentBlockId());
+      } else {
+        LOG.info("Close or cancel throws IOExecption {}.", e.getMessage());
```

Comment:
My concern is that, I am not sure if the caller of `closeOrCancelCacheStream` making any assumption here: is it possible that previously the caller assume exceptions thrown to indicate the error, while now this assumption is no more true? Please double-check if callers of `closeOrCancelCacheStream` did make this assumption.